clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -1460,10 +1460,14 @@ public void handle(OffsetFetchResponse response, RequestFuture<Map<TopicPartitio
                 if (responseError == Errors.COORDINATOR_LOAD_IN_PROGRESS) {
                     // just retry
                     future.raise(responseError);
-                } else if (responseError == Errors.NOT_COORDINATOR) {
+                } else if (responseError == Errors.COORDINATOR_NOT_AVAILABLE ||
+                           responseError == Errors.NOT_COORDINATOR) {
                     // re-discover the coordinator and retry
                     markCoordinatorUnknown(responseError);
                     future.raise(responseError);
+                } else if (responseError.exception() instanceof RetriableException) {
```

Comment:
Would it make sense to move this branch to after `} else if (responseError == Errors.GROUP_AUTHORIZATION_FAILED) {` to ensure that we handle all the special cases before we handle the general ones?

Reply:
Makes sense, I'll reorder the branches.